Makefile

```diff
@@ -1,58 +1,156 @@
-bin-karmada-dashboard-web:
-	BUILD_PLATFORMS=$(GOOS)/$(GOARCH) hack/build.sh karmada-dashboard-web
-image-karmada-dashboard-web:
-	BUILD_PLATFORMS=linux/$(GOARCH) hack/build.sh karmada-dashboard-web
-	cp -R ui/apps/dashboard/dist _output/bin/linux/$(GOARCH)/dist
-	DOCKER_FILE="build-web.Dockerfile" VERSION=$(VERSION) REGISTRY=$(REGISTRY) BUILD_PLATFORMS=linux/$(GOARCH) hack/docker.sh karmada-dashboard-web
 
+###################
+# Dependencies    #
+###################
+
+# Install Go dependencies
+.PHONY: install-deps
+install-deps:
+	go mod tidy
+	go mod verify
+
+# Install UI dependencies
+.PHONY: install-ui-deps
+install-ui-deps:
+	cd ui && pnpm install
+
+# Install all dependencies
+.PHONY: install
+install: install-deps install-ui-deps
+
+###################
+# Development     #
+###################
+
+# Run both API and Web server
+.PHONY: run
+run:
+ifndef KUBECONFIG
+	$(error KUBECONFIG is required. Please specify the path to karmada kubeconfig)
+endif
+	@echo "Starting API server..."
+	@_output/bin/$(GOOS)/$(GOARCH)/karmada-dashboard-api \
+		--karmada-kubeconfig=$(KUBECONFIG) \
+		--karmada-context=$(KARMADA_CTX) \
+		--kubeconfig=$(KUBECONFIG) \
+		--context=$(HOST_CTX) \
+		--insecure-port=$(API_PORT) & echo $$! > .api.pid
+	@echo "Starting Web server..."
+	@cd ui && VITE_API_HOST=$(API_HOST) pnpm run dashboard:dev || (kill `cat .api.pid` && rm .api.pid)
+	@if [ -f .api.pid ]; then kill `cat .api.pid` && rm .api.pid; fi
+
+# Run API server only
+.PHONY: run-api
+run-api:
+ifndef KUBECONFIG
+	$(error KUBECONFIG is required. Please specify the path to karmada kubeconfig)
+endif
+	_output/bin/$(GOOS)/$(GOARCH)/karmada-dashboard-api \
+		--karmada-kubeconfig=$(KUBECONFIG) \
+		--karmada-context=$(KARMADA_CTX) \
+		--kubeconfig=$(KUBECONFIG) \
+		--context=$(HOST_CTX) \
+		--insecure-port=$(API_PORT)
+
+# Run Web server only
+.PHONY: run-web
+run-web:
+	cd ui && VITE_API_HOST=$(API_HOST) pnpm run dashboard:dev
+
+###################
+# Help            #
+###################
+
+# Show help
+.PHONY: help
+help:
+	@echo "Karmada Dashboard Makefile"
+	@echo ""
+	@echo "Usage:"
+	@echo "  make              - Build all binaries (same as 'make all')"
+	@echo "  make all          - Build all binaries"
+	@echo "  make install      - Install all dependencies"
+	@echo "  make run          - Run both API and Web servers"
+	@echo "  make images       - Build all Docker images"
+	@echo ""
+	@echo "Development Commands:"
+	@echo "  make run-api      - Run API server only"
+	@echo "  make run-web      - Run Web server only"
+	@echo "  make install-deps - Install Go dependencies"
+	@echo "  make install-ui-deps - Install UI dependencies"
+	@echo ""
+	@echo "Build Commands:"
+	@echo "  make build        - Build all binaries"
+	@echo "  make images       - Build all Docker images"
+	@echo ""
+	@echo "Variables:"
+	@echo "  KUBECONFIG       - Path to karmada kubeconfig file"
+	@echo "  API_PORT         - API server port (default: 8000)"
```

Comment:
should we also display `KARMADA_CTX` 、`HOST_CTX` here?

Reply:
updated.